engine/access/rpc/backend/backend_transactions.go

```diff
@@ -1039,3 +1040,292 @@ func (b *backendTransactions) tryGetTransactionResultByIndex(
+					Hex("block_id", req.GetBlockId()).
+					Uint32("index", req.GetIndex()).
+					Msg("Successfully got transaction error message by index from any node")
+				return nil
+			}
+			return err
+		},
+		nil,
+	)
+	if errToReturn != nil {
+		b.log.Err(errToReturn).Msg("failed to get transaction error message by index from execution nodes")
+	}
+
+	return resp, errToReturn
+}
+
+// getTransactionErrorMessagesFromAnyEN performs an RPC call using available nodes passed as argument.
+func (b *backendTransactions) getTransactionErrorMessagesFromAnyEN(
+	ctx context.Context,
+	execNodes flow.IdentityList,
+	req *execproto.GetTransactionErrorMessagesByBlockIDRequest,
+) ([]*execproto.GetTransactionErrorMessagesResponse_Result, error) {
+	var errToReturn error
+
+	// if we were passed 0 execution nodes add a specific error
+	if len(execNodes) == 0 {
+		return nil, errors.New("zero execution nodes")
+	}
+
+	var resp *execproto.GetTransactionErrorMessagesResponse
+	errToReturn = b.nodeCommunicator.CallAvailableNode(
+		execNodes,
+		func(node *flow.Identity) error {
+			var err error
+			resp, err = b.tryGetTransactionErrorMessagesByBlockIDFromEN(ctx, node, req)
+			if err == nil {
+				b.log.Debug().
+					Str("execution_node", node.String()).
+					Hex("block_id", req.GetBlockId()).
+					Msg("Successfully got transaction error messages from any node")
+				return nil
+			}
+			return err
+		},
+		nil,
+	)
+
+	// log the errors
+	if errToReturn != nil {
+		b.log.Err(errToReturn).Msg("failed to get transaction error messages from execution nodes")
+	}
+
+	return resp.GetResults(), errToReturn
+}
+
+// tryGetTransactionErrorMessageFromEN performs a grpc call to the specified execution node and returns response.
+func (b *backendTransactions) tryGetTransactionErrorMessageFromEN(
+	ctx context.Context,
+	execNode *flow.Identity,
+	req *execproto.GetTransactionErrorMessageRequest,
+) (*execproto.GetTransactionErrorMessageResponse, error) {
+	execRPCClient, closer, err := b.connFactory.GetExecutionAPIClient(execNode.Address)
+	if err != nil {
+		return nil, err
+	}
+	defer closer.Close()
+
+	return execRPCClient.GetTransactionErrorMessage(ctx, req)
+}
+
+// tryGetTransactionErrorMessageByIndexFromEN performs a grpc call to the specified execution node and returns response.
+func (b *backendTransactions) tryGetTransactionErrorMessageByIndexFromEN(
+	ctx context.Context,
+	execNode *flow.Identity,
+	req *execproto.GetTransactionErrorMessageByIndexRequest,
+) (*execproto.GetTransactionErrorMessageResponse, error) {
+	execRPCClient, closer, err := b.connFactory.GetExecutionAPIClient(execNode.Address)
+	if err != nil {
+		return nil, err
+	}
+	defer closer.Close()
+	return execRPCClient.GetTransactionErrorMessageByIndex(ctx, req)
+}
+
+// tryGetTransactionErrorMessagesByBlockIDFromEN performs a grpc call to the specified execution node and returns response.
+func (b *backendTransactions) tryGetTransactionErrorMessagesByBlockIDFromEN(
```

Comment:
```suggestion
// Expected errors during normal operation:
//   - status.Error - GRPC call failed, some of possible codes are:
//      - codes.NotFound - request cannot be served by EN because of absence of data.
//      - codes.Unavailable - remote node is not unavailable.
func (b *backendTransactions) tryGetTransactionErrorMessagesByBlockIDFromEN(
```